projects/js-packages/publicize-components/src/components/post-publish-share-status/index.tsx

```diff
@@ -1,29 +1,38 @@
 import { useSelect } from '@wordpress/data';
 import { PluginPostPublishPanel } from '@wordpress/edit-post';
+import { store as editorStore } from '@wordpress/editor';
 import { store as socialStore } from '../../social-store';
-import { ShareStatusModal } from '../share-status-modal';
+import { ShareStatus } from './share-status';
 
 /**
  * Post publish share status component.
  *
  * @return {import('react').ReactNode} - Post publish share status component.
  */
 export function PostPublishShareStatus() {
-	const { featureFlags } = useSelect( select => {
+	const { featureFlags, postId, isPostPublised, willPostBeShared } = useSelect( select => {
 		const store = select( socialStore );
+
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- `@wordpress/editor` is a nightmare to work with TypeScript
+		const _editorStore = select( editorStore ) as any;
+
+		const meta = _editorStore.getEditedPostAttribute( 'meta' );
+
 		return {
 			featureFlags: store.featureFlags(),
+			postId: _editorStore.getCurrentPostId(),
+			isPostPublised: _editorStore.isCurrentPostPublished(),
+			willPostBeShared: meta.jetpack_publicize_feature_enabled,
 		};
 	}, [] );
 
-	if ( ! featureFlags.useShareStatus ) {
+	if ( ! featureFlags.useShareStatus || ! willPostBeShared || ! isPostPublised ) {
```

Comment:
Nitpick: Should we also return in the case of Atomic and Simple sites? Resharing for simple sites can happen only if they have a paid plan. We can tweak these in a follow-up as well since the feature flag controls the feature.

Reply:
I think feature flag should control that. I fact, we should hide the whole `<PostPublishShareStatus />` behind the feature flag inside [`PostPublishPanels`](https://github.com/Automattic/jetpack/blob/04769556004833774f2caa652761dbb99898e9a4/projects/js-packages/publicize-components/src/components/post-publish-panels/index.tsx#L6). CC @gmjuhasz

Reply:
Do we do that next or in here?

Reply:
I can do that in #38952, what do you say?

Reply:
Sounds good!